Implementation of bluetooth transport in the SDL

nk0leg opened the pull request:
see detail in [SDLWIN-386](https://adc.luxoft.com/jira/browse/SDLWIN-384)

Commit:
Add Bluetooth utils for parsing and stores bluetooth service record

Commit:
Add Connect method for WinNative socket implementation

nk0leg commented:
@malirod , @anosach-luxoft  please review

anosach-luxoft commented on src/components/utils/include/utils/bluetooth_win/bluetooth_service_record.h:
```diff
@@ -0,0 +1,150 @@
+   *
+   * @return True if service_uuid and stored UUID equil.
+   */
+  bool IsUuidEquil(const std::string& service_uuid) const {
```
@nk0leg `equil` looks like mistake

nk0leg replied:
see f045cd6

anosach-luxoft commented on src/components/utils/src/socket_win.cc:
```diff
@@ -99,6 +99,8 @@ class utils::TcpSocketConnection::Impl {
 
   bool Connect(const HostAddress& address, const uint16_t port);
 
+  bool Connect(const SOCKET bth_soket);
```
@nk0leg do not use `bth` prefix here. This method could be used not only for BT

nk0leg replied:
see f045cd6

anosach-luxoft commented:
Reviewed

Commit:
Add bluetooth device and service scanner implementation

nk0leg commented:
@malirod  please look 7e6d24d

Commit:
Update the implementation of device scanner and a services scanner

Commit:
Add interfaces for compatible with Qt

Commit:
Fix on Bluetooth Transport adapter of save state
Windows bluetooth API not used rfcomm_chanel for connect to SPT.
If we saves rfcomm_chanel and trying connects to SPT after SDl reboot -
we get connection error.
Because after reboot SDL tries connect to SPT using rfcomm_chanel from app_info.data

nk0leg commented:
see #169
